### PR TITLE
Migrate citation -> cr:citeAs

### PR DESCRIPTION
**Pull request by ccl-core**

### CODE REVIEW

File: python/mlcroissant/recipes/croissant.json [hunk 1]
```diff
@@ -0,0 +1,115 @@
+{
```
Comment (marcenacp) [inline python/mlcroissant/recipes/croissant.json:1]: Remove this file?

File: python/mlcroissant/mlcroissant/scripts/migrations/previous/202302061400.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+"""Migration: citation should be citeAs."""
+
+def up(json_ld):
+    """Up migration to set conformsTo to croissant specs 1.0."""
+    if "citation" in json_ld:
+        json_ld["citeAs"] = json_ld["citation"]
+        del json_ld["citation"]
+    return json_ld
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/scripts/migrations/previous/202302061400.py]: Trailing space

File: python/mlcroissant/mlcroissant/_src/structure_graph/nodes/metadata.py [hunk 3]
```diff
@@ -259,6 +259,12 @@ def from_jsonld(cls, ctx: Context, metadata: Json) -> Metadata:
         ctx.conforms_to = CroissantVersion.from_jsonld(
             ctx, metadata.get(constants.DCTERMS_CONFORMS_TO)
         )
+
+        if ctx.is_v0:
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/structure_graph/nodes/metadata.py]: `ctx.is_v0` -> `ctx.is_v0()`

File: editor/views/metadata.py [hunk 4]
```diff
@@ -97,14 +97,14 @@ def _render_generic_metadata(metadata: Metadata):
         on_change=handle_metadata_change,
         args=(MetadataEvent.LICENSE, metadata, key),
     )
-    key = "metadata-citation"
+    key = "metadata-cite-as"
     st.text_area(
-        label="Citation",
+        label="Cite as",
```
Comment (marcenacp) [inline editor/views/metadata.py:102]: As a label, I find `Citation` more readable

Comment (ccl-core) [reply editor/views/metadata.py]: SG, Done

**Comment by ccl-core** [comment]
This addresses #490

**Comment by ccl-core** [comment]
The notebook tests fail for TFDS `CroissantBuilder`, which is trying to get `metadata.citation` instead of `metadata.cite_as` to populate the `citation` field in its `DatasetInfo`. I will merge this PR despite the failing tests, and will fix the `CroissantBuilder` from TFDS side.

**Pull request merged**